jetty-util/src/main/java/org/eclipse/jetty/util/Pool.java

```diff
@@ -281,17 +326,12 @@ public boolean isClosed()
     public void close()
     {
         List<Entry> copy;
-        lock.lock();
-        try
+        try (Locker.Lock l = locker.lock())
         {
             closed = true;
             copy = new ArrayList<>(sharedList);
             sharedList.clear();
         }
-        finally
-        {
-            lock.unlock();
-        }
 
         // iterate the copy and close its entries
         for (Entry entry : copy)
```

Comment:
Can we use `IO.close()` in the loop? I think a `warn()` here is too much.

Reply:
Sure.... but should we close any Closeable when it is removed from the pool, or only when the pool is closed?  Perhaps we should close AutoCloseables that are removed from the pool.  Eitherway, need to javadoc this behavior